### DISCUSSION TITLE
Second Auth Provider connection attached to existing user

### DISCUSSION THREAD

**Discussion by levino** [pocketbase/pocketbase#5782] [Q&A] [answered]
I did the following:

I created a new user by using `.authWithOAuth2` with google. Then I opened an **incognito window** and called `.authWithOAuth2` again with github. Both users (google and github) use the same e-mail address. To my surprise, the second user creation did not fail, but succeeded and github was added as an authorized provider to the existing user with the e-mail address.

I find this behaviour concerning since it effectively allows someone to take over my account if they only use my e-mail address with an auth provider which my app supports but where I did not create an account yet. Am I overlooking something? Is it possible to disable this behaviour? I would rather expect that the second `.authWithOAuth2` call would fail as long as I am not already logged in as the user.

Thank you for clarification.

**Comment by ganigeorgiev** [comment, marked as answer]
Linking by email is a standard practice with OAuth2. There is a big caveat though - we allow linking by email ONLY if the provider's account **has a verified email address**.

OAuth2 (and any other 3rd party auth) relies on trust and in this case we trust that the provider's email verification works correctly. If you can't trust it, then don't use the provider.

If you really want to prevent a single user to have more than one OAuth2 account linked then you can attach to the [`OnRecordBeforeAuthWithOAuth2Request` hook](https://pocketbase.io/docs/go-event-hooks/#onrecordbeforeauthwithoauth2request) and perform a lookup for the other available linked auths in the `_externalAuths` table.
